Build failure: libraspberrypi

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

`nix build github:NixOS/nixpkgs#legacySystems.aarch64-linux.libraspberrypi`



### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/318045378

### Relevant log output

```console
[ 52%] Building C object interface/vchiq_arm/CMakeFiles/vchiq_arm.dir/vchiq_lib.c.o
In file included from /build/source/build/inc/interface/vcos/vcos.h:144,
                 from /build/source/interface/vchi/vchi_mh.h:31,
                 from /build/source/interface/vchiq_arm/vchiq_if.h:31,
                 from /build/source/interface/vchiq_arm/vchiq.h:31,
                 from /build/source/interface/vchiq_arm/vchiq_lib.c:33:
/build/source/interface/vchiq_arm/vchiq_lib.c: In function 'vchiq_release_message':
/build/source/interface/vchiq_arm/vchiq_lib.c:401:77: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]
  401 |    vcos_log_trace( "%s handle=%08x, header=%x", __func__, (uint32_t)handle, (uint32_t)header );
      |                                                                             ^
/build/source/build/inc/interface/vcos/vcos_logging.h:234:113: note: in definition of macro '_VCOS_LOG_X'
  234 | #  define _VCOS_LOG_X(cat, _level, fmt...)   do { if (vcos_is_log_enabled(cat,_level)) vcos_log_impl(cat,_level,fmt); } while (0)
      |                                                                                                                 ^~~
/build/source/interface/vchiq_arm/vchiq_lib.c:401:4: note: in expansion of macro 'vcos_log_trace'
  401 |    vcos_log_trace( "%s handle=%08x, header=%x", __func__, (uint32_t)handle, (uint32_t)header );
      |    ^~~~~~~~~~~~~~
/build/source/interface/vchiq_arm/vchiq_lib.c: At top level:
/build/source/interface/vchiq_arm/vchiq_lib.c:534:1: error: conflicting types for 'vchiq_bulk_receive_handle'; have 'VCHIQ_STATUS_T(VCHIQ_SERVICE_HANDLE_T,  VCHI_MEM_HANDLE_T,  void *, int,  void *, VCHIQ_BULK_MODE_T,  int (*)(void))' {aka 'VCHIQ_STATUS_T(unsigned int,  int,  void *, int,  void *, VCHIQ_BULK_MODE_T,  int (*)(void))'}
  534 | vchiq_bulk_receive_handle(VCHIQ_SERVICE_HANDLE_T handle,
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/interface/vchiq_arm/vchiq_if.h:174:23: note: previous declaration of 'vchiq_bulk_receive_handle' with type 'VCHIQ_STATUS_T(VCHIQ_SERVICE_HANDLE_T,  VCHI_MEM_HANDLE_T,  void *, int,  void *, VCHIQ_BULK_MODE_T,  int (*)(char *, const struct pagelist_struct *))' {aka 'VCHIQ_STATUS_T(unsigned int,  int,  void *, int,  void *, VCHIQ_BULK_MODE_T,  int (*)(char *, const struct pagelist_struct *))'}
  174 | extern VCHIQ_STATUS_T vchiq_bulk_receive_handle(VCHIQ_SERVICE_HANDLE_T service,
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/interface/vchiq_arm/vchiq_lib.c:610:1: warning: conflicting types for 'vchiq_use_service' due to enum/integer mismatch; have 'int32_t(const VCHIQ_SERVICE_HANDLE_T)' {aka 'int(const unsigned int)'} [-Wenum-int-mismatch]
  610 | vchiq_use_service( const VCHIQ_SERVICE_HANDLE_T handle )
      | ^~~~~~~~~~~~~~~~~
/build/source/interface/vchiq_arm/vchiq_if.h:150:23: note: previous declaration of 'vchiq_use_service' with type 'VCHIQ_STATUS_T(VCHIQ_SERVICE_HANDLE_T)' {aka 'VCHIQ_STATUS_T(unsigned int)'}
  150 | extern VCHIQ_STATUS_T vchiq_use_service(VCHIQ_SERVICE_HANDLE_T service);
      |                       ^~~~~~~~~~~~~~~~~
/build/source/interface/vchiq_arm/vchiq_lib.c:623:1: warning: conflicting types for 'vchiq_release_service' due to enum/integer mismatch; have 'int32_t(const VCHIQ_SERVICE_HANDLE_T)' {aka 'int(const unsigned int)'} [-Wenum-int-mismatch]
  623 | vchiq_release_service( const VCHIQ_SERVICE_HANDLE_T handle )
      | ^~~~~~~~~~~~~~~~~~~~~
/build/source/interface/vchiq_arm/vchiq_if.h:151:23: note: previous declaration of 'vchiq_release_service' with type 'VCHIQ_STATUS_T(VCHIQ_SERVICE_HANDLE_T)' {aka 'VCHIQ_STATUS_T(unsigned int)'}
  151 | extern VCHIQ_STATUS_T vchiq_release_service(VCHIQ_SERVICE_HANDLE_T service);
      |                       ^~~~~~~~~~~~~~~~~~~~~
/build/source/interface/vchiq_arm/vchiq_lib.c: In function 'vchiq_dump_phys_mem':
/build/source/interface/vchiq_arm/vchiq_lib.c:1399:31: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
 1399 |    VCHIQ_SERVICE_T *service = (VCHIQ_SERVICE_T *)handle;
      |                               ^
/build/source/interface/vchiq_arm/vchiq_lib.c: In function 'completion_thread':
/build/source/interface/vchiq_arm/vchiq_lib.c:1552:36: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]
 1552 |                completion->reason, (uint32_t)completion->header,
      |                                    ^
/build/source/build/inc/interface/vcos/vcos_logging.h:234:113: note: in definition of macro '_VCOS_LOG_X'
  234 | #  define _VCOS_LOG_X(cat, _level, fmt...)   do { if (vcos_is_log_enabled(cat,_level)) vcos_log_impl(cat,_level,fmt); } while (0)
      |                                                                                                                 ^~~
/build/source/interface/vchiq_arm/vchiq_lib.c:1551:13: note: in expansion of macro 'vcos_log_trace'
 1551 |             vcos_log_trace( "callback(%x, %x, %x(%x,%x), %x)",
      |             ^~~~~~~~~~~~~~
/build/source/interface/vchiq_arm/vchiq_lib.c:1553:16: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]
 1553 |                (uint32_t)&service->base, (uint32_t)service->lib_handle, (uint32_t)service->base.userdata, (uint32_t)completion->bulk_userdata );
      |                ^
/build/source/build/inc/interface/vcos/vcos_logging.h:234:113: note: in definition of macro '_VCOS_LOG_X'
  234 | #  define _VCOS_LOG_X(cat, _level, fmt...)   do { if (vcos_is_log_enabled(cat,_level)) vcos_log_impl(cat,_level,fmt); } while (0)
      |                                                                                                                 ^~~
/build/source/interface/vchiq_arm/vchiq_lib.c:1551:13: note: in expansion of macro 'vcos_log_trace'
 1551 |             vcos_log_trace( "callback(%x, %x, %x(%x,%x), %x)",
      |             ^~~~~~~~~~~~~~
/build/source/interface/vchiq_arm/vchiq_lib.c:1553:73: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]
 1553 |                (uint32_t)&service->base, (uint32_t)service->lib_handle, (uint32_t)service->base.userdata, (uint32_t)completion->bulk_userdata );
      |                                                                         ^
/build/source/build/inc/interface/vcos/vcos_logging.h:234:113: note: in definition of macro '_VCOS_LOG_X'
  234 | #  define _VCOS_LOG_X(cat, _level, fmt...)   do { if (vcos_is_log_enabled(cat,_level)) vcos_log_impl(cat,_level,fmt); } while (0)
      |                                                                                                                 ^~~
/build/source/interface/vchiq_arm/vchiq_lib.c:1551:13: note: in expansion of macro 'vcos_log_trace'
 1551 |             vcos_log_trace( "callback(%x, %x, %x(%x,%x), %x)",
      |             ^~~~~~~~~~~~~~
/build/source/interface/vchiq_arm/vchiq_lib.c:1553:107: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]
 1553 |                (uint32_t)&service->base, (uint32_t)service->lib_handle, (uint32_t)service->base.userdata, (uint32_t)completion->bulk_userdata );
      |                                                                                                           ^
/build/source/build/inc/interface/vcos/vcos_logging.h:234:113: note: in definition of macro '_VCOS_LOG_X'
  234 | #  define _VCOS_LOG_X(cat, _level, fmt...)   do { if (vcos_is_log_enabled(cat,_level)) vcos_log_impl(cat,_level,fmt); } while (0)
      |                                                                                                                 ^~~
/build/source/interface/vchiq_arm/vchiq_lib.c:1551:13: note: in expansion of macro 'vcos_log_trace'
 1551 |             vcos_log_trace( "callback(%x, %x, %x(%x,%x), %x)",
      |             ^~~~~~~~~~~~~~
```

### Additional context

`libraspberrypi` is no longer being developed, with most of the functionality either being available through standard APIs or https://github.com/raspberrypi/tools (which seems not packaged). perhaps it might be appropriate to drop it? a few nixpkgs packages depend on it in non-default configs

- Potenitally related: #475479


### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.2, NixOS, 26.05 (Yarara), 26.05pre-git`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.95.0
System type: x86_64-linux
Additional system types: aarch64-linux, armv7l-linux, i686-linux, powerpc-linux, powerpc64-linux, powerpc64le-linux, riscv32-linux, riscv64-linux, wasm32-wasi
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/darkkirb/.config/nix/nix.conf:/nix/store/6yjcpmq1r7fzrm6wm0zi94vw7mms1dpn-stylix-kde-config/nix/nix.conf:/home/darkkirb/.config/kdedefaults/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/darkkirb/.local/share/flatpak/exports/etc/xdg/nix/nix.conf:/var/lib/flatpak/exports/etc/xdg/nix/nix.conf:/home/darkkirb/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/darkkirb/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/darkkirb/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/jzbyj8ayfr8xq99h2qanxb6iapszh68z-lix-2.95.0/share`
 - nixpkgs: `/etc/nix/inputs/nixpkgs`

### Notify maintainers

@dezgeg 
@tkerber 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/li/libraspberrypi/package.nix
+++ b/pkgs/by-name/li/libraspberrypi/package.nix
@@ -22,6 +22,8 @@ stdenv.mkDerivation {
     pkg-config
   ];
 
+  env.NIX_CFLAGS_COMPILE = "-std=gnu17";
+
   cmakeFlags = [
     # -DARM64=ON disables all targets that only build on 32-bit ARM; this allows
     # the package to build on aarch64 and other architectures

